src/components/buoy-telemetry/BuoyMap.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <div class="mapboxgl-map-container">
+    <div class="date is-hidden-touch is-size-4">
+      <span>{{ formattedDate }}</span>
+    </div>
+    <div class="date-mobile is-hidden-desktop is-size-5">
+      <span>{{ formattedDate }}</span>
+    </div>
+    <div ref="el" class="mapbox-container" />
+  </div>
+
+  <!-- reference image for mapbox to pull from -->
+  <img
+    ref="imageEl"
+    :src="BuoyMarker"
+    style="
+      visibility: hidden;
+      height: 15px;
+      width: 15px;
+      position: absolute;
+      top: 0;
+    "
+  />
+</template>
+
+<script setup lang="ts">
+import { computed, inject, onMounted, ref, watch } from "vue";
+import mapboxgl from "mapbox-gl";
+import { scaleSqrt, scaleLinear } from "d3-scale";
+
+import BuoyMarker from "@/assets/illustrations/buoy-marker.svg";
+import { Data } from "../../utils/erddap";
+import { BuoyStore } from "../../store/buoy";
+
+interface Sample extends Data {
+  date: Date;
+}
+
+const store = inject("store") as BuoyStore;
+
+const props = defineProps<{
+  samples: Sample[];
+  formattedDate: string;
+  variable: string;
+}>();
+
+const annotatedSamples = computed(() => {
+  const domain = [0, 45];
```

Comment:
I had commented on old PR, but that one was merged already 🙃. Can we abstract the embedded constants? Something like
```
const clorophyll_domain = [0,45]
const clorophyll_range...
const clorophyll_colors
```

You're probably going to get more variables in here soon and keeping the abtracted and labeling them to match the variable they represent will keep it more readable. 

Also are the domain and range decided by visual inspection? Expert advice? We should probably keep in mind for future legend

Reply:
I abstracted to `varDomain`, `sizeRange`, and `colorRange` consts in dictionaries so we can add in other variables later. For now I have them in three dicts, but maybe it should be one nested dict?